tests/distributions/test_transformed_distribution.py

```diff
@@ -51,30 +51,42 @@ def inverse(self, y, *args, **kwargs):
         x = (torch.log(y) - self.b) / self.a
         return x
 
-    def log_det_jacobian(self, y, *args, **kwargs):
+    def batch_log_det_jacobian(self, y, *args, **kwargs):
         """
-        Calculates the determinant of the log jacobian
+        Calculates the elementwise determinant of the log jacobian
         """
-        return torch.log(torch.abs(self.a)) + torch.log(y)
+        return (torch.log(torch.abs(self.a)) + torch.log(y)).sum(-1).unsqueeze(-1)
 
 
-@pytest.fixture()
-def lognormal():
+def make_lognormal(kwargs):
+    kwargs['examples'] = [kwargs.pop('example')]
     return Fixture(pyro_dist=(dist.lognormal, LogNormal),
                    scipy_dist=sp.lognorm,
-                   examples=[
-                       {'mu': [1.4], 'sigma': [0.4],
-                        'test_data': [5.5]},
-                       {'mu': [1.4], 'sigma': [0.4],
-                        'test_data': [[5.5]]},
-                       {'mu': [[1.4, 0.4, 0.4], [1.4, 0.4, 0.6]],
-                        'sigma': [[2.6, 0.5, 0.5], [2.6, 0.5, 0.5]],
-                        'test_data': [[5.5, 6.4, 6.4], [5.5, 6.4, 6.4]]},
-                       {'mu': [[1.4], [0.4]], 'sigma': [[2.6], [0.5]],
-                        'test_data': [[5.5], [6.4]]}
-                   ],
                    scipy_arg_fn=lambda mu, sigma: ((np.array(sigma),),
-                                                   {"scale": np.exp(np.array(mu))}))
+                                                   {"scale": np.exp(np.array(mu))}),
+                   **kwargs)
+
+
+EXAMPLES = list(map(make_lognormal, [
+    {
+        'example': {'mu': [1.4], 'sigma': [0.4], 'test_data': [5.5]},
+    },
+    {
+        'example': {'mu': [1.4], 'sigma': [0.4], 'test_data': [[5.5]]},
+    },
+    {
+        'example': {
+            'mu': [1.4, 0.4, 0.4],
```

Comment:
I think my comment got lost somewhere:
 - modified shapes of mu/sigma so that we are not sending multiple params for the batch dimension (to get the right shape, and also calculating variance wouldn't make sense otherwise).
 - the lognormal distribution seems to be very sensitive to sigma, so dialed down its value and upped the number of samples to get the analytic means/variances to match up.